src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -4728,6 +4728,8 @@ void CodeGen::genIntrinsic(GenTreeIntrinsic* treeNode)
     emitAttr size = emitActualTypeSize(op1);
     bool     is4  = (size == EA_4BYTE);
 
+    bool isUnsigned = (treeNode->gtFlags & GTF_UNSIGNED) != 0;
```

Comment:
It would be best to introduce dedicated intrinsic names (for internal use only) for the unsigned variants of these instructions instead of using `GTF_UNSIGNED`. Then you won't need some of the VN changes, which currently look incomplete because both variants get the same VNFunc.